fiftyone/core/map/process.py

```diff
@@ -0,0 +1,302 @@
+            worker_progress = True
+            progress = False
+            lock = ctx.RLock()
+            tqdm.set_lock(lock)
+        else:
+            worker_progress = False
+            lock = None
+
+        queue = multiprocessing.Queue()
+        batch_count = multiprocessing.Value("i", 0)
+        sample_count = (
+            multiprocessing.Value("i", 0) if progress is not False else None
+        )
+        cancel_event = multiprocessing.Event()
+
+        # Extract information from sample collection.
+        if isinstance(sample_collection, fov.DatasetView):
+            # pylint:disable-next=protected-access
+            dataset_name = sample_collection._root_dataset.name
+            # pylint:disable-next=protected-access
+            view_stages = sample_collection._serialize()
+        else:
+            dataset_name = sample_collection.name
+            view_stages = None
+
+        sample_batches = self._batch_cls.split(
+            sample_collection, self.workers, self.batch_size
+        )
+
+        pool = ctx.Pool(
+            processes=self.workers,
+            initializer=_init_worker,
+            initargs=(
+                dataset_name,
+                view_stages,
+                pickle.dumps(map_fcn),
+                batch_count,
+                sample_count,
+                queue,
+                cancel_event,
+                save,
+                worker_progress,
+                skip_failures,
+                lock,
+            ),
+        )
+
+        pb = fou.ProgressBar(
+            total=sum(batch.total for batch in sample_batches),
+            progress=progress,
+        )
+
+        num_batches = len(sample_batches)
+        with pool, pb:
+            pool.map_async(
+                _map_batch,
+                (
+                    (idx + 1, num_batches, batch)
+                    for idx, batch in enumerate(sample_batches)
+                ),
+            )
+
+            sample_errors: List[Tuple[bson.ObjectId, Exception, None]] = []
+
+            # Initialize backoff parameters
+            initial_timeout = 0.1
+            max_timeout = 5.0
+            backoff_factor = 2
+            current_timeout = initial_timeout
+
+            while True:
+                try:
+                    sample_id, err, result = queue.get(timeout=current_timeout)
+                    # Reset backoff on successful get
+                    current_timeout = initial_timeout
+                except Empty:
+                    # Apply exponential backoff, but cap at max_timeout
+                    current_timeout = min(
+                        current_timeout * backoff_factor, max_timeout
+                    )
+
+                    # Reset backoff if we've hit too many consecutive timeouts
+                    # This prevents getting stuck with very long timeouts
+                    if current_timeout == max_timeout:
+                        current_timeout = initial_timeout
```

Comment:
why would it go back to the initial timeout instead of just staying at the max timeout? Seems a little weird to keep looping back around

Reply:
Sure - I thought going back to an initial time out is nice so that we don't have to wait at max time out